Source/VMware.vSphereDSC/DSCResources/VMHostNetwork/VMHostVss/VMHostVssSecurity.ps1

```diff
@@ -0,0 +1,168 @@
+
+    <#
+    .DESCRIPTION
+
+    The flag to indicate whether or not the Media Access Control (MAC) address
+    can be changed.
+    #>
+    [DscProperty()]
+    [boolean] $MacChanges
+
+    [void] Set() {
+        Write-Verbose -Message "$(Get-Date) $($s = Get-PSCallStack; "Entering {0}" -f $s[0].FunctionName)"
+
+        $this.ConnectVIServer()
+        $vmHost = $this.GetVMHost()
+        $this.GetNetworkSystem($vmHost)
+
+        $this.UpdateVssSecurity($vmHost)
+    }
+
+    [bool] Test() {
+        Write-Verbose -Message "$(Get-Date) $($s = Get-PSCallStack; "Entering {0}" -f $s[0].FunctionName)"
+
+        $this.ConnectVIServer()
+        $vmHost = $this.GetVMHost()
+        $this.GetNetworkSystem($vmHost)
+        $vss = $this.GetVss($vmHost)
+
+        if ($this.Ensure -eq [Ensure]::Present) {
+            return ($null -ne $vss -and $this.Equals($vss))
+        }
+        else {
+            $this.AllowPromiscuous = $false
+            $this.ForgedTransmits = $true
+            $this.MacChanges = $true
+
+            return ($null -eq $vss -or $this.Equals($vss))
+        }
+    }
+
+    [VMHostVssSecurity] Get() {
+        Write-Verbose -Message "$(Get-Date) $($s = Get-PSCallStack; "Entering {0}" -f $s[0].FunctionName)"
+
+        $result = [VMHostVssSecurity]::new()
+        $result.Server = $this.Server
+
+        $this.ConnectVIServer()
+        $vmHost = $this.GetVMHost()
+        $this.GetNetworkSystem($vmHost)
+        $result.Name = $vmHost.Name
+        $this.PopulateResult($vmHost, $result)
+
+        if ($null -ne $result.VssName) {
+            $result.Ensure = 'Present'
+        }
+        else {
+            $result.Ensure = 'Absent'
+        }
+
+        return $result
+    }
+
+    <#
+    .DESCRIPTION
+
+    Returns a boolean value indicating if the VMHostVssSecurity should to be updated.
+    #>
+    [bool] Equals($vss) {
+        Write-Verbose -Message "$(Get-Date) $($s = Get-PSCallStack; "Entering {0}" -f $s[0].FunctionName)"
+
+        $vssSecurityTest = @()
+        $vssSecurityTest += ($vss.Spec.Policy.Security.AllowPromiscuous -eq $this.AllowPromiscuous)
+        $vssSecurityTest += ($vss.Spec.Policy.Security.ForgedTransmits -eq $this.ForgedTransmits)
+        $vssSecurityTest += ($vss.Spec.Policy.Security.MacChanges -eq $this.MacChanges)
+
+        return ($vssSecurityTest -notcontains $false)
+    }
+
+    <#
+    .DESCRIPTION
+
+    Updates the configuration of the virtual switch.
+    #>
+    [void] UpdateVssSecurity($vmHost) {
+        Write-Verbose -Message "$(Get-Date) $($s = Get-PSCallStack; "Entering {0}" -f $s[0].FunctionName)"
+
+        $vssSecurityArgs = @{
+            Name = $this.VssName
+            AllowPromiscuous = $this.AllowPromiscuous
+            ForgedTransmits = $this.ForgedTransmits
+            MacChanges = $this.MacChanges
+        }
+        $vss = $this.GetVss($vmHost)
+
+        if ($this.Ensure -eq 'Present') {
+            if ($this.Equals($vss)) {
```

Comment:
This could be simplified to:

```
 if (!$this.Equals($vss)) {
     $vssSecurityArgs.Add('Operation', 'edit')
 }
```

Reply:
But then it will hit the Update-Network at the end, which is not desired.

Reply:
Agree.